### ISSUE TITLE
iOS 13.4 Beta Crash

### ISSUE THREAD

**Issue by johnlietzke** [xmartlabs/Eureka#1989]
Eureka 5.2 Cocoapods Installation

No compiler or installation errors and the project runs.
On calling a Eureka FormViewController I receive multiple crashes but the most consistent is:

public var cell: Cell! {
        return _cell ?? {
            let result = cellProvider.makeCell(style: self.cellStyle)
            result.row = self
            result.setup()
            _cell = result
            return _cell
        }()
    }

The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKitCore/UIView.h> may also be helpful.

**Comment by mats-claassen** [comment]
Hi @johnlietzke What is the crash? And the stack trace?

**Comment by johnlietzke** [comment]
Hi Mathias,

Thanks for the response.

The error occurs at:

 let result = cellProvider.makeCell(style: self.cellStyle)

#2	0x0000000103932e22 in closure #1 in implicit closure #1 in Row.cell.getter at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Pods/Eureka/Source/Core/Row.swift:155


Is this what you are looking?

If not let me know what you else you need?

Best,
John

**Comment by lch88** [comment]
Hi, I'm also having the exact same issue.

Here's more detailed information. I'm using Eureka 5.2 via SPM.

Crash:
```
Thread 1: EXC_BAD_ACCESS (code=EXC_I386_GPFLT)
```

Stack trace:
```
Thread 1 Queue : com.apple.main-thread (serial)
#0	0x00007fff50c65c67 in objc_retain ()
#1	0x000000010f669a0f in Row.cell.getter at /Users/<redacted>/SourcePackages/checkouts/Eureka/Source/Core/Row.swift:154
#2	0x000000010f66aab1 in protocol witness for TypedRowType.cell.getter in conformance Row<A> ()
#3	0x000000010f671aaa in closure #1 in RowType<>.cellUpdate(_:) at /Users/<redacted>/SourcePackages/checkouts/Eureka/Source/Core/RowType.swift:216
#4	0x000000010f66a637 in Row.updateCell() at /Users/<redacted>/SourcePackages/checkouts/Eureka/Source/Core/Row.swift:178
#5	0x000000010f63c255 in FormViewController.tableView(_:cellForRowAt:) at /Users/<redacted>/SourcePackages/checkouts/Eureka/Source/Core/Core.swift:969
#6	0x000000010f63c465 in @objc FormViewController.tableView(_:cellForRowAt:) ()
#7	0x00007fff48fbe657 in -[UITableView _createPreparedCellForGlobalRow:withIndexPath:willDisplay:] ()
#8	0x00007fff48f876f8 in -[UITableView _updateVisibleCellsNow:] ()
#9	0x00007fff48fa77a5 in -[UITableView layoutSubviews] ()
#10	0x00007fff4929a5b8 in -[UIView(CALayerDelegate) layoutSublayersOfLayer:] ()
#11	0x00007fff2b7bca80 in -[CALayer layoutSublayers] ()
#12	0x00007fff2b7c2c0b in CA::Layer::layout_if_needed(CA::Transaction*) ()
#13	0x00007fff2b7ce2c2 in CA::Layer::layout_and_display_if_needed(CA::Transaction*) ()
#14	0x00007fff2b7173ac in CA::Context::commit_transaction(CA::Transaction*, double) ()
#15	0x00007fff2b74ad87 in CA::Transaction::commit() ()
#16	0x00007fff48de0054 in _afterCACommitHandler ()
#17	0x00007fff23db91c7 in __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ ()
#18	0x00007fff23db3c7e in __CFRunLoopDoObservers ()
#19	0x00007fff23db41ca in __CFRunLoopRun ()
#20	0x00007fff23db39e4 in CFRunLoopRunSpecific ()
#21	0x00007fff38dc9c1a in GSEventRunModal ()
#22	0x00007fff48db6408 in UIApplicationMain ()
#23	0x000000010e77c6a3 in main at /Users/<redacted>/main.swift:32
#24	0x00007fff51ace10d in start ()
#25	0x00007fff51ace10d in start ()
```

I hope this is useful.

**Comment by mats-claassen** [comment]
`Thread 1: EXC_BAD_ACCESS (code=EXC_I386_GPFLT)` is the only error message? 
That doesn't say much. 

As this is still beta it could be an iOS issue. Mainly because there is no apparent error and this code was working up to now. Let's wait if a future beta fixes it.

**Comment by johnlietzke** [comment]
This is the stack trace I am receving.

Thread 1 Queue : com.apple.main-thread (serial)
#0	0x00007fff51c0c33a in __pthread_kill ()
#1	0x00007fff51cb6e60 in pthread_kill ()
#2	0x00007fff51b9b85c in abort ()
#3	0x00007fff513f3b75 in swift::fatalError(unsigned int, char const*, ...) ()
#4	0x00007fff513f3d62 in swift::swift_abortRetainUnowned(void const*) ()
#5	0x00007fff513f562a in swift_unownedRetainStrong ()
#6	0x0000000102cfc25b in closure #1 in RowType<>.cellSetup(_:) at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Pods/Eureka/Source/Core/RowType.swift:227
#7	0x0000000102cfc31f in thunk for @escaping @callee_guaranteed (@guaranteed A.Cell) -> () ()
#8	0x0000000102cfc391 in partial apply for thunk for @escaping @callee_guaranteed (@guaranteed A.Cell) -> () ()
#9	0x0000000102cf3ccc in thunk for @escaping @callee_guaranteed (@in_guaranteed A) -> (@out ()) ()
#10	0x0000000102cf3917 in Row._cell.didset at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Pods/Eureka/Source/Core/Row.swift:148
#11	0x0000000102cf3b55 in Row._cell.setter ()
#12	0x0000000102cf3f58 in closure #1 in implicit closure #1 in Row.cell.getter at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Pods/Eureka/Source/Core/Row.swift:158
#13	0x0000000102cf3dbb in Row.cell.getter at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Pods/Eureka/Source/Core/Row.swift:154
#14	0x0000000102cf4053 in Row.baseCell.getter at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Pods/Eureka/Source/Core/Row.swift:164
#15	0x0000000102c5f3c4 in FormViewController.tableView(_:estimatedHeightForRowAt:) at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Pods/Eureka/Source/Core/Core.swift:797
#16	0x0000000102c5f727 in @objc FormViewController.tableView(_:estimatedHeightForRowAt:) ()
#17	0x00007fff48fcbbb9 in -[UITableView _estimatedHeightForRowAtIndexPath:] ()
#18	0x00007fff48fdc1cf in __66-[UISectionRowData refreshWithSection:tableView:tableViewRowData:]_block_invoke ()
#19	0x00007fff48fdb848 in -[UISectionRowData refreshWithSection:tableView:tableViewRowData:] ()
#20	0x00007fff48fdd635 in -[UITableViewRowData _ensureSectionOffsetIsValidForSection:] ()
#21	0x00007fff48fe0f96 in -[UITableViewRowData rectForFooterInSection:heightCanBeGuessed:] ()
#22	0x00007fff48fe1209 in -[UITableViewRowData heightForTable] ()
#23	0x00007fff48f7d0ae in -[UITableView _updateContentSizeSkippingContentOffsetAdjustment:] ()
#24	0x00007fff48fab722 in -[UITableView _rebuildGeometryForcingRowDataUpdate:skipContentOffsetAdjustment:updateImmediatelyIfPossible:] ()
#25	0x00007fff48fa93ab in -[UITableView setLayoutMargins:] ()
#26	0x00007fff48fa922d in -[UITableView _setDefaultLayoutMargins:] ()
#27	0x00007fff48fa9177 in -[UITableView didMoveToWindow] ()
#28	0x00007fff4929102c in -[UIView(Internal) _didMoveFromWindow:toWindow:] ()
#29	0x00007fff49209ce7 in -[UIScrollView _didMoveFromWindow:toWindow:] ()
#30	0x00007fff49290aa0 in -[UIView(Internal) _didMoveFromWindow:toWindow:] ()
#31	0x00007fff492843e5 in __45-[UIView(Hierarchy) _postMovedFromSuperview:]_block_invoke ()
#32	0x00007fff492842ea in -[UIView(Hierarchy) _postMovedFromSuperview:] ()
#33	0x00007fff49293868 in -[UIView(Internal) _addSubview:positioned:relativeTo:] ()
#34	0x00007fff4876e715 in -[_UIParallaxDimmingView didMoveToWindow] ()
#35	0x00007fff4929102c in -[UIView(Internal) _didMoveFromWindow:toWindow:] ()
#36	0x00007fff49290aa0 in -[UIView(Internal) _didMoveFromWindow:toWindow:] ()
#37	0x00007fff492843e5 in __45-[UIView(Hierarchy) _postMovedFromSuperview:]_block_invoke ()
#38	0x00007fff492842ea in -[UIView(Hierarchy) _postMovedFromSuperview:] ()
#39	0x00007fff49293868 in -[UIView(Internal) _addSubview:positioned:relativeTo:] ()
#40	0x00007fff48769ca9 in __53-[_UINavigationParallaxTransition animateTransition:]_block_invoke_2 ()
#41	0x00007fff4928c7a4 in +[UIView(Animation) performWithoutAnimation:] ()
#42	0x00007fff487693e7 in __53-[_UINavigationParallaxTransition animateTransition:]_block_invoke ()
#43	0x00007fff49291ff6 in +[UIView(Internal) _performBlockDelayingTriggeringResponderEvents:forScene:] ()
#44	0x00007fff48768d9b in -[_UINavigationParallaxTransition animateTransition:] ()
#45	0x00007fff4875d342 in ___UIViewControllerTransitioningRunCustomTransition_block_invoke_2 ()
#46	0x00007fff48876df8 in +[UIInputResponderController _pinInputViewsForInputResponderController:onBehalfOfResponder:duringBlock:] ()
#47	0x00007fff4875d2d1 in ___UIViewControllerTransitioningRunCustomTransition_block_invoke.645 ()
#48	0x00007fff4928c4cf in +[UIView(Animation) _setAlongsideAnimations:toRunByEndOfBlock:] ()
#49	0x00007fff4875d157 in _UIViewControllerTransitioningRunCustomTransition ()
#50	0x00007fff4867efc3 in -[UINavigationController _startCustomTransition:] ()
#51	0x00007fff486944b1 in -[UINavigationController _startDeferredTransitionIfNeeded:] ()
#52	0x00007fff486958a0 in -[UINavigationController __viewWillLayoutSubviews] ()
#53	0x00007fff48676870 in -[UILayoutContainerView layoutSubviews] ()
#54	0x00000001041c6cc8 in -[UILayoutContainerViewAccessibility layoutSubviews] ()
#55	0x00007fff4929a5b8 in -[UIView(CALayerDelegate) layoutSublayersOfLayer:] ()
#56	0x00007fff2b7bca80 in -[CALayer layoutSublayers] ()
#57	0x00007fff2b7c2c0b in CA::Layer::layout_if_needed(CA::Transaction*) ()
#58	0x00007fff2b7ce2c2 in CA::Layer::layout_and_display_if_needed(CA::Transaction*) ()
#59	0x00007fff2b7173ac in CA::Context::commit_transaction(CA::Transaction*, double) ()
#60	0x00007fff2b74ad87 in CA::Transaction::commit() ()
#61	0x00007fff48daf9e4 in _UIApplicationFlushRunLoopCATransactionIfTooLate ()
#62	0x00007fff48e535bf in __handleEventQueueInternal ()
#63	0x00007fff48e4afa2 in __handleHIDEventFetcherDrain ()
#64	0x00007fff23db9dd1 in __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ ()
#65	0x00007fff23db9cfc in __CFRunLoopDoSource0 ()
#66	0x00007fff23db94d4 in __CFRunLoopDoSources0 ()
#67	0x00007fff23db40ce in __CFRunLoopRun ()
#68	0x00007fff23db39e4 in CFRunLoopRunSpecific ()
#69	0x00007fff38dc9c1a in GSEventRunModal ()
#70	0x00007fff48db6408 in UIApplicationMain ()
#71	0x00000001012f039b in main at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Transit/AppDelegate.swift:10
#72	0x00007fff51ace10d in start ()
#73	0x00007fff51ace10d in start ()

**Comment by mats-claassen** [comment]
Is that in beta 1 or 2? Does it happen in beta 2?

**Comment by johnlietzke** [comment]
It happens in Beta 1, 2 & 3. The last trace I sent you was from beta 3.

**Comment by johnlietzke** [comment]
> Thread 1 Queue : com.apple.main-thread (serial)
#0	0x0000000194bc0e30 in initializeNonMetaClass ()
#1	0x0000000194bc2318 in initializeAndMaybeRelock(objc_class*, objc_object*, mutex_tt<false>&, bool) ()
#2	0x0000000194bcf074 in lookUpImpOrForward ()
#3	0x0000000194bbbe9c in _objc_msgSend_uncached ()
#4	0x000000019523ff7c in -[NSKeyValueSlowMutableArray insertObject:atIndex:] ()
#5	0x00000001951d0864 in -[NSKeyValueNotifyingMutableArray insertObject:atIndex:] ()
#6	0x0000000104229ca0 in Section.append(_:) at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Pods/Eureka/Source/Core/Section.swift:286
#7	0x00000001041ed5d4 in <<< infix(_:_:) at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Pods/Eureka/Source/Core/Operators.swift:124
#8	0x00000001024ff74c in RouteInputController.setUpTableView() at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Transit/New Group/Input/RouteInputController.swift:511
#9	0x00000001024f89bc in RouteInputController.loadView() at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Transit/New Group/Input/RouteInputController.swift:38
#10	0x00000001024f89fc in @objc RouteInputController.loadView() ()
#11	0x00000001988fab20 in -[UIViewController loadViewIfRequired] ()
#12	0x00000001988fb224 in -[UIViewController view] ()
#13	0x000000019885ab40 in -[UINavigationController _startCustomTransition:] ()
#14	0x000000019886ea08 in -[UINavigationController _startDeferredTransitionIfNeeded:] ()
#15	0x000000019886fda8 in -[UINavigationController __viewWillLayoutSubviews] ()
#16	0x0000000198853308 in -[UILayoutContainerView layoutSubviews] ()
#17	0x00000001993b5264 in -[UIView(CALayerDelegate) layoutSublayersOfLayer:] ()
#18	0x000000019b936a04 in -[CALayer layoutSublayers] ()
#19	0x000000019b93cb5c in CA::Layer::layout_if_needed(CA::Transaction*) ()
#20	0x000000019b9476a0 in CA::Layer::layout_and_display_if_needed(CA::Transaction*) ()
#21	0x000000019b890798 in CA::Context::commit_transaction(CA::Transaction*, double) ()
#22	0x000000019b8ba128 in CA::Transaction::commit() ()
#23	0x000000019b8bab3c in CA::Transaction::observer_callback(__CFRunLoopObserver*, unsigned long, void*) ()
#24	0x0000000194e1b628 in __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ ()
#25	0x0000000194e1651c in __CFRunLoopDoObservers ()
#26	0x0000000194e16998 in __CFRunLoopRun ()
#27	0x0000000194e162a4 in CFRunLoopRunSpecific ()
#28	0x000000019eefe38c in GSEventRunModal ()
#29	0x0000000198f20bf4 in UIApplicationMain ()
#30	0x000000010272cf6c in main at /Users/johnlietzke/Library/Mobile Documents/com~apple~CloudDocs/Xcode Projects/Transit/Transit/AppDelegate.swift:10
#31	0x0000000194c9d8f0 in start ()
>

**Comment by johnlietzke** [comment]
This is from iOS 13.4 beta 4 using Xcode 11.4 beta 3.  It appears to be slightly different than the pervious Stack Trace.

**Comment by LeoAiolia** [comment]
ios 13.4 crash, stable version, 
![图片](https://user-images.githubusercontent.com/16910025/77522125-acca4480-6ebe-11ea-95ef-24befd54e38e.png)

**Comment by redlock** [comment]
Facing the same issue with latest XCode 11.4
It is triggered when cellUpdate is used on a row.

**Comment by tanpengsccd** [comment]
while change Xcode to 11.3.1  , cellUpdate  won't crash

**Comment by mats-claassen** [comment]
Duplicate of #1988 #1987 #1999 

Will be fixed in 5.2.1 released soon!
